Implement Strategy Picker

sfmskywalker opened the pull request:
This PR adds UI to allow the user to select a resilience strategy for resilient activities such as SendHttpRequest.

For those activities, a new tab is added called *Resilience*

<img width="1267" alt="image" src="https://github.com/user-attachments/assets/f655bd0d-15ff-4b86-8b77-0bc5a61b7da6" />

sfmskywalker committed:
Add resilience strategy support in workflow activity editor

This commit introduces support for managing resilience strategies in the workflow activity editor, including a new `ResilienceTab` component and supporting services like `IResilienceStrategyCatalog`. It removes the obsolete `SerializerOptions` and updates dependencies to their latest versions, ensuring compatibility and enhanced functionality.

sfmskywalker committed:
Update ResilienceTab to use dynamic resilience category

Refactor ResilienceTab to support a configurable resilience category, defaulting to "Default". Update logic to retrieve the category from activity descriptors and apply it when listing resilience strategies. Remove unused using directive for cleaner code.

sfmskywalker committed:
Add and integrate ExpressionEditor with async content updates

Introduced a `SetContentAsync` method in `ExpressionEditor` for handling asynchronous content updates in the Monaco editor. Enhanced the `ResilienceTab` component to initialize the `ExpressionEditor` and correctly update its content based on resilience strategy configuration. Adjusted other minor code areas for consistency and error handling.

raymonddenhaan commented on Directory.Packages.props:
```diff
@@ -4,26 +4,26 @@
-        <PackageVersion Include="Elsa.Integrations.Agents.Models" Version="$(ElsaIntegrationsVersion)"/>
-        <PackageVersion Include="Elsa.Api.Client" Version="$(ElsaVersion)"/>
-        <PackageVersion Include="Elsa.Secrets.Models" Version="$(ElsaVersion)"/>
+        <PackageVersion Include="Elsa.Integrations.Agents.Models" Version="3.5.0-preview.20"/>
```
Would it not be better to keep using the variables?

sfmskywalker replied:
Yes, if it worked consistently. Oftentimes, Rider (or .NET, I am not sure), older packages are being resolved. I'll revert this at some point to using variables again.

raymonddenhaan commented on src/modules/Elsa.Studio.Workflows.Core/Domain/Services/RemoteResilienceStrategyCatalog.cs:
```diff
@@ -0,0 +1,19 @@
+    public async ValueTask<IEnumerable<JsonObject>> ListAsync(string category, CancellationToken cancellationToken = default)
+    {
+        var api = await backendApiClientProvider.GetApiAsync<IResilienceStrategiesApi>(cancellationToken);
+        var response = await api.LisAsync(category, cancellationToken);
```
typo in the API being used here -> ListAsync

raymonddenhaan commented on src/modules/Elsa.Studio.Workflows/Elsa.Studio.Workflows.csproj:
```diff
@@ -21,5 +21,9 @@
     </ItemGroup>
+
+    <ItemGroup>
+      <Folder Include="Shared\Serialization\" />
```
If this folder is empty, please remove it

sfmskywalker committed:
Fix typo in method call and update package references.

Corrected a typo in the method `ListAsync` in `RemoteResilienceStrategyCatalog`. Updated package references to use dynamic version properties and bumped `ElsaVersion` to `3.5.0-preview.2948`. Also replaced project references with package references in `Elsa.Studio.Core.csproj`.

sfmskywalker committed:
Remove unused Shared\Serialization folder reference

The Shared\Serialization folder reference was redundant and has been removed from the project file. This cleanup improves project maintainability and eliminates unnecessary clutter.
